docs/proposals/14-reversed-cluster-vpn.md

```diff
@@ -0,0 +1,276 @@
+
+**Note:** The network mesh spanned by OpenVPN uses the network range `192.168.123.0 - 192.168.123.255`. This network range cannot be used in either shoot-, or seed clusters. If it is used this might cause subtle problem due to network range overlaps. Unfortunately, this appears not to be well documented but this restriction exists since the very beginning. We should clean up this technical debt as part of the excercise.
+
+### Goals
+
+* We intend to supersede the current VPN solution with the solution outlined in this proposal.
+* We intend to remove the code for the konnectivity tunnel once this solution proposal has been validated.
+
+### Non Goals
+
+* The solution is not a low latency, or high throughput solution. As the kube-apiserver to shoot cluster traffic does not demand these properties we do not intend to invest in improvements.
+* We do not intend to provide continuous availability to the shoot-seed VPN connection. We expect the availability to be comparable to the existing solution.
+
+## Proposal
+
+The proposal is depicted in the following diagram:
+
+![alt text](assets/ReversedTunnelVPN.png "Reversed Tunnel VPN")
+
+We have added an OpenVPN server pod (`vpn-seed-server`) to each control plane. The OpenVPN client in the shoot cluster (`vpn-shoot-client`) connects to the OpenVPN server.
+
+The two containers `vpn-seed-server` and `vpn-shoot-client` are new containers and are not related to containers in the github.com/gardener/vpn project. We will create a new project github.com/gardener/vpn2 for these containers. With this solution we intend to supersede the containers from the github.com/gardener/vpn project.
+
+A service `vpn-seed-server` of type `ClusterIP` is created for each control plane in its namespace.
+
+The `vpn-shoot-client` pod connects to the correct `vpn-seed-server` service via the SNI passthrough proxy introduced with [SNI Passthrough proxy for kube-apiservers](https://github.com/gardener/gardener/blob/master/docs/proposals/08-shoot-apiserver-via-sni.md) port 8132. 
+
+Shoot OpenVPN clients (`vpn-shoot-client`) connect to the correct OpenVPN Server using the http proxy feature provided by OpenVPN. A configuration is added to the envoy proxy to detect http proxy requests and forward the connection attempt to the correct OpenVPN server.
+
+The `kube-apiserver` to shoot cluster connections are established using the API server proxy feature via an envoy proxy sidecar container of the `vpn-seed-server` container. 
+
+The restriction regarding the `192.168.123.0/24` network range in the current VPN solution still applies to this proposal. No other restrictions are introduced. In the context of this GEP a pull requst has been filed to block usage of that range by shoot clusters.
```

Comment:
Can't we (ab)use the [IPV4 address ranges](https://tools.ietf.org/html/rfc5737#section-3) used for documentation instead of the Private IP address ranges?

Reply:
No, I would not go for the documentation address ranges. This is from RFC 5737:

```
4.  Operational Implications

   Addresses within the TEST-NET-1, TEST-NET-2, and TEST-NET-3 blocks
   SHOULD NOT appear on the public Internet and are used without any
   coordination with IANA or an Internet registry [RFC2050].  Network
   operators SHOULD add these address blocks to the list of non-
   routeable address spaces, and if packet filters are deployed, then
   this address block SHOULD be added to packet filters.

   These blocks are not for local use, and the filters may be used in
   both local and public contexts.
   ```